netket/graph/graph.py

```diff
@@ -0,0 +1,166 @@
+from abstract_graph import AbstractGraph
+
+import numpy as _np
+import networkx as _nx
+
+
+class NetworkX(AbstractGraph):
+    """ Wrapper for a networkx graph"""
+
+    def __init__(self, graph):
+        """
+        Constructs a netket graph from a networkx graph.
+
+        Args: 
+            graph: A networkx graph (might be a networkx.Graph or a networkx.MultiGraph)
+        Examples:
+            A graph of nodes [0,1,2] with edges [(0,1), (0,2), (1,2)]
+            >>> import netket
+            >>> import networkx
+            >>> nx_g = networkx.Graph([(0,1), (0,2), (1,2)])
+            >>> nk_g = netket.graph.NetworkX(nx_g)
+            >>> print(nk_g.n_nodes)
+            3
+        """
+        assert isinstance(graph, _nx.classes.graph.Graph) or isinstance(
+            graph, _nx.classes.multigraph.MultiGraph
+        )
+
+        if isinstance(graph, _nx.classes.graph.Graph):
+            self.graph = _nx.MultiGraph(graph)
+
+        self._automorphisms = None
+
+        super().__init__()
+
+    @property
+    def adjacency_list(self):
+        return [list(self.graph.neighbors(node)) for node in self.graph.nodes]
+
+    @property
+    def is_connected(self):
+        # TODO: how to check if a multigraph is connected?
+        return _nx.is_connected(self.graph)
+
+    def edges(self, color=False):
+        if color is True:
+            return list(self.graph.edges(keys=True))
+        elif color is not False:
+            return [(u, v) for u, v, k in self.graph.edges if k == color]
+        else:
+            return list(self.graph.edges(keys=False))
+
+    def distances(self):
+        return _nx.floyd_warshall_numpy(self.graph)
+
+    @property
+    def is_bipartite(self):
+        return _nx.is_bipartite(self.graph)
+
+    @property
+    def n_nodes(self):
+        return self.graph.number_of_nodes()
+
+    def automorphisms(self):
+        # TODO: check how to compute these when we have a coloured graph where there could
+        #       be a duplicated edge with two different colors.
+
+        # For the moment, if there are colors, the method returns a NotImplementedError:
+        if self.edges():
+            colors = _np.unique(_np.array(self.edges(color=True))[:, 2])
+        else:
+            colors = _np.array([])
+        if colors.size >= 2:
+            raise NotImplementedError
```

Comment:
I think this should have an error message.
```suggestion
            raise NotImplementedError("automorphisms is not yet implemented for colored edges")
```